ui/app/actions.js

```diff
@@ -449,6 +460,26 @@ function signTx (txData) {
   }
 }
 
+function estimateGas ({ to, amount }) {
+  return (dispatch) => {
```

Comment:
Can you make this return a promise that resolve with data?

We can simply call this in the component, and update gas that way. Since the gas is specific to one transaction, we don't have to make it a redux state.